includes/class-wpcom-legacy-redirector.php

```diff
@@ -13,27 +13,45 @@ class WPCOM_Legacy_Redirector {
 	const POST_TYPE   = 'vip-legacy-redirect';
 	const CACHE_GROUP = 'vip-legacy-redirect-2';
 
+	// Plugin version.
+	const PLUGIN_VERSION = '1.3.0';
+
+	// Used to flip the version of the available roles capabilities.
+	const CAPABILITIES_VER = '1.0';
+
 	/**
 	 * Actions and filters.
 	 */
 	static function start() {
 		add_action( 'init', array( __CLASS__, 'init' ) );
+		add_action( 'admin_init', array( __CLASS__, 'admin_init' ) );
 		add_filter( 'template_redirect', array( __CLASS__, 'maybe_do_redirect' ), 0 ); // hook in early, before the canonical redirect.
 		add_action( 'admin_menu', array( new WPCOM_Legacy_Redirector_UI(), 'admin_menu' ) );
 		add_filter( 'admin_enqueue_scripts', array( __CLASS__, 'wpcom_legacy_add_redirect_js' ) );
 		add_filter( 'bulk_actions-edit-' . Post_Type::POST_TYPE, array( __CLASS__, 'remove_bulk_edit' ) );
 	}
 
+	/**
+	 * Initialize and register other classes.
+	 */
+	static function admin_init() {
+		$capabilities_version_key = self::POST_TYPE . '_capabilities_version';
+
+		if ( self::CAPABILITIES_VER <= get_option( $capabilities_version_key ) ) {
```

Comment:
I think this logic is wrong. Currently it says if the constant is less then or equal to the saved version number (which it should be for much of the time), then register the new capability.

I _think_ this should be:

```suggestion
		if ( self::CAPABILITIES_VER > get_option( $capabilities_version_key ) ) {
```

The integration tests below check that the capability is added to the roles, but it doesn't check how many times this `Capability->register()` command is run, which is part of what we're  trying to fix here.

Reply:
found a way to test this, I've moved related code to the capability class and added a return statement in that function, this would give us an easy way to test the register call after the initial init